data/mods/ssb/abilities.ts

```diff
@@ -37,7 +37,7 @@ export function changeSet(context: Battle, pokemon: Pokemon, newSet: SSBSet, cha
 	pokemon.maxhp = newMaxHP;
 	let item = newSet.item;
 	if (typeof item !== 'string') item = item[Math.floor(Math.random() * item.length)];
-	pokemon.setItem(item);
+	if (context.toID(item) !== pokemon.item) pokemon.setItem(item);
```

Comment:
shouldnt this be `===`? otherwise it would still regenerate jho's item

Reply:
we only want to change the item if the new item (`item`) is different from the old item (`pokemon.item`)

Reply:
yes, but when jho consumes his item, `pokemon.item` becomes `''`, right? so it would still regenerate

Reply:
Oh. Is there a way to tell what the item was before it was consumed? (Checking `if (pokemon.item)` seems too naive.)

Reply:
`this.lastItem`?

Reply:
sure